fdbclient/include/fdbclient/MetaclusterManagement.actor.h

```diff
@@ -647,6 +646,15 @@ struct RegisterClusterImpl {
 	ACTOR static Future<Void> registerInManagementCluster(RegisterClusterImpl* self,
 	                                                      Reference<typename DB::TransactionT> tr) {
 		state Optional<DataClusterMetadata> dataClusterMetadata = wait(tryGetClusterTransaction(tr, self->clusterName));
+		state Reference<IDatabase> dataClusterDb = wait(openDatabase(self->connectionString));
+		state Reference<ITransaction> dataTr = dataClusterDb->createTransaction();
+		dataTr->setOption(FDBTransactionOptions::ACCESS_SYSTEM_KEYS);
+		state Optional<MetaclusterRegistrationEntry> currentMetaclusterRegistration =
+		    wait(MetaclusterMetadata::metaclusterRegistration().get(dataTr));
+		if (!currentMetaclusterRegistration.present() ||
+		    currentMetaclusterRegistration.get().id != self->clusterEntry.id) {
+			throw cluster_removed();
+		}
```

Comment:
I think this still has the same problem, just over a narrower time window, due to the fact that we are doing the verification in a separate transaction.

Instead, I think we could solve this by adding an extra step at the beginning to reserve the specified name for the cluster being added. Then a second cluster couldn't register simultaneously with the first.

You may need the data cluster ID for this, in which case you would have to run another extra step to fetch that ID. On the other hand, there may be value in not using a static data cluster ID, in which case we might not need to do this.

Reply:
I was wrong, it looks like the data cluster ID is generated for each registration. That means we shouldn't need an extra step to fetch it, and you can instead generate it when your are creating a `REGISTERING` entry for the data cluster in the management cluster.